fastlane/spec/runner_spec.rb

```diff
@@ -32,5 +33,84 @@
         end
       end
     end
+
+    describe "#execute" do
+      before do
+        @ff = Fastlane::FastFile.new('./fastlane/spec/fixtures/fastfiles/FastfileLaneKeywordParams')
+      end
+
+      context 'when a lane does not expect any parameter' do
+        it 'accepts calling the lane with no parameter' do
+          result = @ff.runner.execute(:lane_no_param, :ios)
+          expect(result).to eq('No parameter')
+        end
+
+        it 'accepts calling the lane with arbitrary (unused) parameter' do
+          result = @ff.runner.execute(:lane_no_param, :ios, { unused1: 42, unused2: true })
+          expect(result).to eq('No parameter')
+        end
+      end
+
+      context 'when a lane expects its parameters as a Hash' do
+        it 'accepts calling the lane with no parameter at all' do
+          result = @ff.runner.execute(:lane_hash_param, :ios)
+          expect(result).to eq('name: nil; version: nil; interactive: nil')
+        end
+
+        it 'accepts calling the lane with less parameters than used by the lane' do
+          result = @ff.runner.execute(:lane_hash_param, :ios, { version: '12.3' })
+          expect(result).to eq('name: nil; version: "12.3"; interactive: nil')
+        end
+
+        it 'accepts calling the lane with more parameters than used by the lane' do
+          result = @ff.runner.execute(:lane_hash_param, :ios, { name: 'test', version: '12.3', interactive: true, unused: 42 })
+          expect(result).to eq('name: "test"; version: "12.3"; interactive: true')
+        end
+      end
+
+      context 'when a lane expects its parameters as keywords' do
+        def keywords_list(list)
+          # The way keyword lists appear in error messages is different in Windows & Linux vs macOS
+          if FastlaneCore::Helper.windows? || FastlaneCore::Helper.linux?
+            list.map(&:to_s).join(', ') # On Windows and Linux, keyword names don't show the `:` prefix in error messages
+          else
+            list.map(&:inspect).join(', ') # On other platforms, they do have `:` in front or keyword names
+          end
+        end
+
+        it 'fails when calling the lane with no parameter at all' do
```

Comment:
Perhaps rephrase this so it's more accurate:

```suggestion
        it 'fails when calling the lane with required parameters not being passed' do
```

Because you could expect e.g. 3 params, all of them have default values, and thus you can call it with no params at all and still succeed 😃